plugins/catalog-react/src/components/EntityPeekAheadPopover/EntityPeekAheadPopover.tsx

```diff
@@ -0,0 +1,207 @@
+ */
+import useAsyncFn from 'react-use/lib/useAsyncFn';
+import { catalogApiRef } from '../../api';
+import React, { PropsWithChildren, useEffect, useState } from 'react';
+import HoverPopover from 'material-ui-popup-state/HoverPopover';
+import {
+  bindHover,
+  bindPopover,
+  usePopupState,
+} from 'material-ui-popup-state/hooks';
+import {
+  Box,
+  Card,
+  CardActions,
+  CardContent,
+  Chip,
+  makeStyles,
+  Tooltip,
+  Typography,
+} from '@material-ui/core';
+import { useApiHolder } from '@backstage/core-plugin-api';
+import {
+  isGroupEntity,
+  isUserEntity,
+  parseEntityRef,
+} from '@backstage/catalog-model';
+import { Progress } from '@backstage/core-components';
+import {
+  EntityCardActions,
+  UserCardActions,
+  GroupCardActions,
+} from './CardActionComponents';
+import { EntityNotFoundCard } from './EntityNotFoundCard';
+import { debounce } from 'lodash';
+
+/**
+ * Properties for an entity popover on hover of a component.
+ *
+ * @public
+ */
+export type EntityPeekAheadPopoverProps = PropsWithChildren<{
+  entityRef: string;
+  delayTime?: number;
+}>;
+
+const useStyles = makeStyles(() => {
+  return {
+    trigger: {
+      display: 'inline-block',
+    },
+    popoverPaper: {
+      width: '30em',
+    },
+    descriptionTypography: {
+      overflow: 'hidden',
+      textOverflow: 'ellipsis',
+      display: '-webkit-box',
+      WebkitLineClamp: 2,
+      WebkitBoxOrient: 'vertical',
+    },
+  };
+});
+
+const maxTagChips = 4;
+
+/**
+ * Shows an entity popover on hover of a component.
+ *
+ * @public
+ */
+export const EntityPeekAheadPopover = (props: EntityPeekAheadPopoverProps) => {
+  const { entityRef, children, delayTime = 500 } = props;
+
+  const classes = useStyles();
+  const apiHolder = useApiHolder();
+  const popupState = usePopupState({
+    variant: 'popover',
+    popupId: 'entity-peek-ahead',
+  });
+  const compoundEntityRef = parseEntityRef(entityRef);
+  const [isHovered, setIsHovered] = useState(false);
+
+  const debouncedHandleMouseEnter = debounce(
+    () => setIsHovered(true),
+    delayTime,
+  );
+
+  const [{ loading, error, value: entity }, load] = useAsyncFn(async () => {
+    const catalogApi = apiHolder.get(catalogApiRef);
+    if (catalogApi) {
+      const retrievedEntity = await catalogApi.getEntityByRef(
+        compoundEntityRef,
+      );
+      if (!retrievedEntity) {
+        throw new Error(`${compoundEntityRef.name} was not found`);
+      }
+      return retrievedEntity;
+    }
+    return undefined;
+  }, [apiHolder, compoundEntityRef]);
+
+  const handleOnMouseLeave = () => {
+    setIsHovered(false);
+    debouncedHandleMouseEnter.cancel();
+  };
+
+  useEffect(() => {
+    if (popupState.isOpen && !entity && !error && !loading) {
+      load();
+    }
+  }, [popupState.isOpen, load, entity, error, loading]);
+
+  return (
+    <>
+      {' '}
+      <span
+        onMouseEnter={debouncedHandleMouseEnter}
+        onMouseLeave={handleOnMouseLeave}
```

Comment:
This doesn't seem to work, did you try it out carefully in the storybook? You can't enter the popover card at all; it closes if you try. If you try to slowly go with the mouse pointer into the test button in the storybook _from below_, the popover never shows.

Reply:
I tried to fix this. Could you take a look now.